server/controllers/notification.js

```diff
@@ -0,0 +1,35 @@
+const Notification = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+
+exports.createNotification = asyncHandler(async (req, res) => {
+    const { notification } = req.body;
```

Comment:
As a good security practice, you should restructure the fields you want to persist in the DB so you won't directly persist whatever the client passes in the request body.

Reply:
I have made the necessary changes; though the controller createNotification method looks similar to before, the req.body is no longer a notification object; instead, it is a json object with the relevant properties as defined in Notification interface in the front end. The necessary restructuring of the fields takes place in the pushnotification method in the notification context which now takes as parameter the relevant properties (type, description, createdBy, etc) instead of a notification object. (this incidentally also makes the method easier to use as now you just need to pass in the properties enclosed in an object instead of importing interface Notification definition and making a notification object to pass in). Please let me know if this is what you are looking for; if not I can change the implementation to whatever you want.